### PR TITLE
Prototype separate save activity

### PR DESCRIPTION
**Pull request by samtstern**

### CODE REVIEW

**Comment by samtstern** [comment]
This is just a basic implementation.  I could clean up some more and fix the tests if we choose to go this way.

File: auth/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,13 +1,12 @@
-<manifest
-    xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    package="com.firebase.ui.auth">
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          xmlns:tools="http://schemas.android.com/tools"
+          package="com.firebase.ui.auth">
```
Comment (SUPERCILEX) [inline auth/src/main/AndroidManifest.xml]: Looks like some weird styling stuff got messed up here and below.

File: auth/src/main/java/com/firebase/ui/auth/ui/credentials/CredentialSaveActivity.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mHandler = ViewModelProviders.of(this).get(SmartLockHandler.class);
+        mHandler.init(getFlowParams());
+
+        Credential credential = getIntent().getParcelableExtra(ExtraConstants.EXTRA_CREDENTIAL);
+        mIdpResponse = getIntent().getParcelableExtra(ExtraConstants.EXTRA_IDP_RESPONSE);
+
+        mHandler.saveCredentials(credential);
```
Comment (SUPERCILEX) [inline auth/src/main/java/com/firebase/ui/auth/ui/credentials/CredentialSaveActivity.java]: This should be guarded by a `savedInstanceState == null` check. (Or should we make the activity resilient to config changes with `android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation"`?)

**Comment by samtstern** [comment]
@SUPERCILEX gonna move this back to #1124

**Pull request merged**